### PR TITLE
(FFM-1814) Fetch flags before starting sse

### PR DESCRIPTION
**Pull request by conormurray95**
**What**
Improve the resiliency of the sdk startup process. 
Startup should be three phases, one after another; authenticate -> pull initial data -> setup sse stream (if enabled) 
Improvements here are:
1. Attempt to authenticate every minute until we succeed - matches java sdk behaviour. Before if auth failed the sdk would silently hang forever 
2. Remove goroutines that fetch initial data and setup sse stream -> this can be handled on the first iteration of the cron job instead, previously the cron and setup functions could start to run over the top of each other if the fetch took too long. Also we shouldn't be starting the stream until we've successfully fetched flags, in the current setup fetching could fail but the stream could connect and we'd never get the initial data - only updates
3. Changed initialized channel to a boolean - theres no need for it to be a channel and it led to deadlocks, plus the isInitialized function could only really be called once which isn't very expected behaviour 

**Why**
The SDK should be more resilient and startup once conditions allow.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -235,83 +242,88 @@ func (c *CfClient) authenticate(ctx context.Context, target evaluation.Target) {
...
 
 func (c *CfClient) makeTicker(interval uint) *time.Ticker {
 	return time.NewTicker(time.Minute * time.Duration(interval))
 }
 
 func (c *CfClient) pullCronJob(ctx context.Context) {
+	poll := func() {
+		c.mux.RLock()
+		if !c.streamConnected {
+			ok := c.retrieve(ctx)
```
Comment (Andrew-Hayes) [inline client/client.go:302]: if we fail to retrieve what do we do?

Comment (conormurray95) [reply client/client.go:302]: We just log the error out in that retrieve function then the next poll interval we try again

Comment (Andrew-Hayes) [reply client/client.go:302]: sweet

**Review by Andrew-Hayes — APPROVED** [review]
LGTM

**Pull request merged**